src/Symfony/Component/DependencyInjection/Compiler/PassConfig.php

```diff
@@ -42,13 +42,13 @@ public function __construct()
         $this->beforeOptimizationPasses = array(
             100 => array(
                 $resolveClassPass = new ResolveClassPass(),
-                new ResolveDefinitionInheritancePass(),
```

Comment:
Thinking about that, this should be kept here: all following compiler passes need to always win. Moving the pass after all DI extension passes is what makes changes tracking a nightmare.
unless I missed something, the full target order proposed in this PR should be:
defaults > instanceof > parents > service > passes